mpqp/core/circuit.py

```diff
@@ -945,9 +945,11 @@ def to_qasm2(self) -> str:
         )
         if TYPE_CHECKING:
             assert isinstance(qiskit_circ, QuantumCircuit)
-        qasm = qiskit_circ.qasm()
-        assert qasm is not None
-        return qasm
+
+        from qiskit import qasm2
+        qasm_str = qasm2.dumps(qiskit_circ)
+        assert qasm_str is not None
```

Comment:
The assert was for the linter, I checked in the `qasm2.dumps` implementation, it can never return None. Either it returns an error if needed, or returns a string, with at least the header "OPENQASM2" in any case.

Reply:
then since `qams2.dumps`'s signature is `QuantumCircuit -> str` (it cannot
return None), I'd remove this line as it serves no purpose

Reply:
Done